src/openforms/js/components/admin/form_design/utils.js

```diff
@@ -57,6 +57,74 @@ const checkKeyChange = (mutationType, newComponent, oldComponent) => {
   return newComponent.key !== oldComponent.key;
 };
 
+const checkIfInitialValueTransformationIsNeeded = (newType, originalVariable) => {
```

Comment:
I can't really picture situations where transformation is NOT needed, except for cases that are already broken like an array of strings but the variable type is set to "string".

Shouldn't we just always reset? Like even from string -> datetime: "foo" is not a valid datetime anyway, but a valid string, so maybe we can cut out some complexity?

Reply:
Mhh, i that thought also crossed my mind.. Maybe it is indeed best to always reset..

The only cases a transformation could be useful/nice is when a value changes to datatype `object` or when a date becomes a datetime (or something simular..). Or a simple value changes to array..

Reply:
let's just keep it simple and always reset then!

Reply:
I've removed it 👍